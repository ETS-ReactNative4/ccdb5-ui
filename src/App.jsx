
import React, { Component } from 'react';
import './App.less';
import ComplaintCard from './ComplaintCard';
import Pagination from './Pagination';
import SearchBar from './SearchBar';
import UrlBarSynch from './UrlBarSynch';

export class App extends Component {
  constructor() {
    super();

    // This binding is necessary to make `this` work in the callback
    // https://facebook.github.io/react/docs/handling-events.html
    this._onPage = this._onPage.bind(this);
    this._onSearch = this._onSearch.bind(this);
    this._onUrlChanged = this._onUrlChanged.bind(this);

    this.urlBar = new UrlBarSynch(this._onUrlChanged);

    // Build up the state from the URL
    let nonQs = {
      items: [],
      total: 0
    };
    let qs = this.urlBar.getParams();

    this.state = Object.assign({}, nonQs, qs);
  }

  componentDidMount() {
    this._callApi();
  }

  render() {
    let to = this.state.from + this.state.size;

    return (
      <main className="col-12">
        <aside className="col-3">
          <div className="refine-panel">Refine Panel</div>
        </aside>
        <div className="col-9 content">
          <SearchBar onSearch={this._onSearch} 
                     searchText={this.state.searchText} />
          <div className="results-panel">
            <summary>
              <Pagination {...this.state} onPage={this._onPage} />
            </summary>
            <ul className="cards-panel">
              {this.state.items
                .filter((e, i) => i >= this.state.from && i < to)
                .map(item => <ComplaintCard key={item.complaint_id}
                                            row={item} />)}
            </ul>
          </div>
        </div>
      </main>
    );
  }

  //---------------------------------------------------------------------------
  // API Call

  _callApi() {
    return fetch('https://data.consumerfinance.gov/resource/jhzv-w97w.json')
    .then(result => result.json())
    .then(items => this.setState({items: items, total: items.length}))
  }

  //---------------------------------------------------------------------------
  // Handlers

  _onSearch(s) {
    let update = {
      searchText: s,
      from: 0,
      total: 0
    };

    this.urlBar.setParams(Object.assign({}, this.state, update));
    this.setState(update);
    this._callApi();
  }

  _onPage(page) {
    let update = {
      from: (page - 1) * this.state.size
    }

    this.urlBar.setParams(Object.assign({}, this.state, update));
    this.setState(update);
  }

  _onUrlChanged(params) {
    this.setState(params);
  }
}
