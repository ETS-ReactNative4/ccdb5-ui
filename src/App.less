@import (less) "~cf-core/src/cf-core.less";
@import (less) "~cf-icons/src/cf-icons.less";
@import (less) "~cf-buttons/src/cf-buttons.less";
@import (less) "~cf-forms/src/cf-forms.less";
@import (less) "~cf-grid/src/cf-grid.less";
@import (less) "~cf-layout/src/cf-layout.less";
@import (less) "~cf-typography/src/cf-typography.less";
@import (less) "~cf-pagination/src/cf-pagination.less";
@import (less) "~cf-expandables/src/cf-expandables.less";
@import (less) "~cf-tables/src/cf-tables.less";

@import (less) "base.less";

// App-wide definitions

.layout-row {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.layout-column {
    display: flex;
    flex-direction: column;
}

.flex-fixed {
    flex: none;
}

.flex-all {
    flex: 1 0 auto;
}

.cols {
    .grid_nested-col-group();
}

.col:before {
    display: block;

    width: 100%;
    content: "";
}

.col-1 {
    .grid_column( 1 );
}

.col-2 {
    .grid_column( 2 );
}

.col-3 {
    .grid_column( 3 );
}

.col-4 {
    .grid_column( 4 );
}

.col-6 {
    .grid_column( 6 );
}

.col-8 {
    .grid_column( 8 );
}

.col-9 {
    .grid_column( 9 );
}

.col-10 {
    .grid_column( 10 );
}

.col-12 {
    .grid_column( 12 );
}

// This component specific

.refine-panel {
    height: 400px;
    padding: @gutter-normal;
    border: 2px dashed @gray-40;
    margin-top: @gutter-normal;
    margin-bottom: @gutter-normal;
}

.content {
    margin-top: @gutter-normal;
    margin-bottom: @gutter-normal;
}

.cards-panel {
    padding: 0;
    margin: 0;
}

summary {
    margin-top: @gutter-normal;
    margin-bottom: @gutter-normal;
}