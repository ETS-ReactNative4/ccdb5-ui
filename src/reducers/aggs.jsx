import { RCV_COMPLAINTS } from '../constants'

export const defaultAggs = {
  timely_response: [
    {key: 'No', active: true, doc_count: 678845},
    {key: 'Yes', active: false, doc_count: 487578}
  ],
  company_response: [
    {key: 'Closed with explanation', active: false, doc_count: 574783},
    {key: 'Closed with monetary relief', active: false, doc_count: 151083},
    {key: 'Closed with non-monetary relief', active: false, doc_count: 94550},
    {key: 'Untimely response', active: false, doc_count: 26894},
    {key: 'Closed', active: false, doc_count: 19151},
    {key: 'Closed without relief', active: false, doc_count: 4263},
    {key: 'Closed with relief', active: false, doc_count: 1347},
    {key: 'In progress', active: false, doc_count: 587},
  ]
}

export default (state = defaultAggs, action) => {
  switch(action.type) {
  case RCV_COMPLAINTS:
    // TODO: Handle translating the buckets to aggs state here
    return state

  default:
    return state
  }
}