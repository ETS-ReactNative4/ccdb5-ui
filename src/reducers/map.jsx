// reducer for the Map Tab
import { STATE_FILTER_ADDED, STATE_FILTER_REMOVED } from '../actions/map'
import {
  STATES_API_CALLED, STATES_FAILED, STATES_RECEIVED
} from '../actions/complaints'
import { TILE_MAP_STATES } from '../constants'

export const defaultState = {
  isLoading: false,
  issue: [],
  product: [],
  selectedState: '',
  state: []
}

export const processAggregations = agg => {
  const total = agg.doc_count
  const chartResults = []
  for ( const k in agg ) {
    if ( agg[k].buckets ) {
      agg[k].buckets.forEach( o => {
        chartResults.push( {
          name: o.key,
          value: o.doc_count,
          pctChange: 1,
          isParent: true,
          hasChildren: false,
          pctOfSet: Math.round( o.doc_count / total * 100 )
            .toFixed( 2 ),
          width: 0.5
        } )
      } )
    }
  }
  return chartResults
}


export const processStateAggregations = agg => {
  const states = Object.values( agg.state.buckets )
    .filter( o => TILE_MAP_STATES.includes( o.key ) )
    .map( o => ( {
      name: o.key,
      value: o.doc_count,
      issue: o.issue.buckets[0].key,
      product: o.product.buckets[0].key
    } ) )

  const stateNames = states.map( o => o.name )

  // patch any missing data
  if ( stateNames.length > 0 ) {
    TILE_MAP_STATES.forEach( o => {
      if ( !stateNames.includes( o ) ) {
        states.push( { name: o, value: 0, issue: '', product: '' } )
      }
    } )
  }
  return states
}

// ----------------------------------------------------------------------------
// Action Handlers

/**
 * Updates the state when an aggregations call is in progress
 *
 * @param {object} state the current state in the Redux store
 * @param {object} action the payload containing the key/value pairs
 * @returns {object} the new state for the Redux store
 */
export function statesCallInProcess( state, action ) {
  return {
    ...state,
    activeCall: action.url,
    isLoading: true
  }
}

/**
 * Expanded logic for handling aggregations returned from the API
 *
 * @param {object} state the current state in the Redux store
 * @param {object} action the payload containing the key/value pairs
 * @returns {object} new state for the Redux store
 */
export function processStatesResults( state, action ) {
  const result = { ...state }

  const stateData = action.data.aggregations.state
  const issueData = action.data.aggregations.issue
  const productData = action.data.aggregations.product
  result.activeCall = ''
  result.isLoading = false
  result.state = processStateAggregations( stateData )
  result.issue = processAggregations( issueData )
  result.product = processAggregations( productData )

  return result
}

/**
 * handling errors from an aggregation call
 *
 * @param {object} state the current state in the Redux store
 * @param {object} action the payload containing the key/value pairs
 * @returns {object} new state for the Redux store
 */
export function processStatesError( state, action ) {
  return {
    ...state,
    activeCall: '',
    issue: [],
    error: action.error,
    isLoading: false,
    product: [],
    state: TILE_MAP_STATES.map( o => ( {
      name: o,
      value: 0,
      issue: '',
      product: ''
    } ) )
  }
}

/**
 * toggle a state map toolbar
 *
 * @param {object} state the current state in the Redux store
 * @returns {object} new state for the Redux store
 */
export function deselectState( state ) {
  return {
    ...state,
    selectedState: ''
  }
}


/**
 * toggle a state map toolbar
 *
 * @param {object} state the current state in the Redux store
 * @param {object} action the payload containing the key/value pairs
 * @returns {object} new state for the Redux store
 */
export function toggleState( state, action ) {
  let { selectedState } = state
  selectedState = selectedState.abbr === action.selectedState.abbr ?
    '' : action.selectedState

  return {
    ...state,
    selectedState
  }
}

// ----------------------------------------------------------------------------
// Action Handlers

/**
 * Creates a hash table of action types to handlers
 *
 * @returns {object} a map of types to functions
 */
export function _buildHandlerMap() {
  const handlers = {}
  handlers[STATES_API_CALLED] = statesCallInProcess
  handlers[STATES_RECEIVED] = processStatesResults
  handlers[STATES_FAILED] = processStatesError
  handlers[STATE_FILTER_ADDED] = toggleState
  handlers[STATE_FILTER_REMOVED] = deselectState

  return handlers
}

const _handlers = _buildHandlerMap()

/**
 * Routes an action to an appropriate handler
 *
 * @param {object} state the current state in the Redux store
 * @param {object} action the command being executed
 * @returns {object} the new state for the Redux store
 */
function handleSpecificAction( state, action ) {
  if ( action.type in _handlers ) {
    return _handlers[action.type]( state, action )
  }

  return state
}

export default ( state = defaultState, action ) => {
  const newState = handleSpecificAction( state, action )
  return newState
}
