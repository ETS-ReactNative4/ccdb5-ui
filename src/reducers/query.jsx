import { SEARCH_CHANGED, PAGE_CHANGED } from '../constants'
export const defaultQuery = {
  searchText: '',
  from: 0,
  size: 10
}

export default (state = defaultQuery, action) => {
  switch(action.type) {
  case SEARCH_CHANGED:
    return {
      ...state,
      searchText: action.searchText,
      from: 0
    }

  case PAGE_CHANGED:
    return {
      ...state,
      from: (action.page - 1) * state.size
    }

  default:
    return state
  }
}
