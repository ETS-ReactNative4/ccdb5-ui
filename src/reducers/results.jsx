/* eslint-disable camelcase */

import {
  API_CALLED, COMPLAINTS_FAILED, COMPLAINTS_RECEIVED
} from '../constants'

const defaultResults = {
  doc_count: 0,
  error: '',
  isLoading: false,
  items: [],
  total: 0
}

export default ( state = defaultResults, action ) => {
  switch ( action.type ) {
    case API_CALLED:
      return {
        ...state,
        isLoading: true
      }

    case COMPLAINTS_RECEIVED: {
      const items = action.data.hits.hits.map( x => x._source )

      const doc_count = Math.max(
      state.doc_count,
      action.data.hits.total,
      action.data._meta.total_record_count
    )

      return {
        ...state,
        doc_count,
        error: '',
        isLoading: false,
        items: items,
        total: action.data.hits.total
      }
    }

    case COMPLAINTS_FAILED:
      return {
        ...defaultResults,
        error: action.error
      }

    default:
      return state
  }
}
