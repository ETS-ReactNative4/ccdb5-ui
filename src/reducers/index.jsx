import { combineReducers } from 'redux'
import aggs from './aggs'
import query from './query'
import results from './results'

export default combineReducers({
    aggs,
    query,
    results
})
