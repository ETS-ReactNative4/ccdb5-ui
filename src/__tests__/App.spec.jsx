import React from 'react';
import { App, DetailComponents } from '../App';
import { defaultQuery } from '../reducers/query'
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

describe('initial state', () => {
  it('renders without crashing', () => {
    defaultQuery.searchText = 'foo';

    const target = renderer.create(
      <MemoryRouter initialEntries={[ '/' ]}>
        <App />
      </MemoryRouter>
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();

  });

  it('renders the detail route', () => {
    const match = { params: { id: '1234' } };
    const detailTarget = renderer.create(
      <MemoryRouter initialEntries={[ '/detail/1234' ]}>
        <DetailComponents match={ match }/>
      </MemoryRouter>
    );

    let detailTree = detailTarget.toJSON();
    expect(detailTree).toMatchSnapshot();
  });
});
