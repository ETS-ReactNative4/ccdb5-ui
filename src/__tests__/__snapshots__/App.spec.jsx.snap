exports[`initial state renders the detail route 1`] = `
<main
  role="main">
  <section
    className="card-container">
    <nav
      className="layout-row">
      <div
        className="back-to-search">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          <span
            className="cf-icon cf-icon-left" />
          Back to search results
        </button>
      </div>
      <div
        className="meaning">
        <a
          href="https://www.consumerfinance.gov/complaint/data-use/"
          target="_blank">
          What do all these data points mean?
        </a>
      </div>
    </nav>
    <section
      className="light-box">
      <div
        className="loading-box">
        <span
          className="cf-icon cf-icon-update cf-icon__before cf-icon__spin" />
        <span>
          This page is loading
        </span>
      </div>
    </section>
  </section>
</main>
`;

exports[`initial state renders without crashing 1`] = `
<main
  className="content content__1-3"
  role="main">
  <header
    className="content_hero">
    <h1
      className="content-header">
      Consumer Complaint Database
    </h1>
    <h4
      className="content-header">
      Beta
    </h4>
    <p>
      Consumer complaints are added to this public database after the company has responded to the complaint, confirming a commercial relationship with the consumer, or after they\'ve had the complaint for 15 calendar days, whichever comes first. We donâ€™t verify all the facts alleged in complaints, but we do give companies the opportunity to publicly respond to complaints by selecting responses from a pre-populated list. Company-level information should be considered in the context of company size and/or market share.
    </p>
    <ul
      className="m-list m-list__horizontal">
      <li
        className="m-list_item">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          More about the complaint database
        </button>
      </li>
      <li
        className="m-list_item">
        <a
          href="https://www.consumerfinance.gov/complaint/data-use/"
          target="_blank">
          How we use complaint data
        </a>
      </li>
      <li
        className="m-list_item">
        <a
          href="https://cfpb.github.io/api/ccdb/"
          target="_blank">
          Technical documentation
        </a>
      </li>
      <li
        className="m-list_item">
        <a
          href="https://data.consumerfinance.gov/dataset/Consumer-Complaints/s6ew-h6mp"
          target="_blank">
          View complaint data in Socrata
        </a>
      </li>
    </ul>
  </header>
  <div
    className="content_wrapper">
    <div
      className="search-panel">
      <h2>
        Search complaint data
      </h2>
      <div>
        <nav
          className="search-bar">
          <form
            action=""
            onSubmit={[Function]}>
            <h5>
              Search Within
            </h5>
            <div
              className="layout-row">
              <div
                className="cf-select">
                <select
                  aria-label="Choose which field will be searched"
                  id="searchField"
                  onChange={[Function]}
                  value="all">
                  <optgroup
                    label="Search Within">
                    <option
                      value="all">
                      All Data
                    </option>
                    <option
                      value="company">
                      Company Name
                    </option>
                    <option
                      value="complaint_what_happened">
                      Narratives
                    </option>
                  </optgroup>
                </select>
              </div>
              <div
                className="flex-all typeahead-portal">
                <section
                  className="typeahead "
                  onBlur={[Function]}
                  onFocus={[Function]}>
                  <input
                    aria-label="The term to search for"
                    autoComplete="off"
                    className="a-text-input"
                    id="searchText"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Enter your search term(s)"
                    type="text"
                    value="foo" />
                </section>
              </div>
              <button
                className="a-btn"
                type="submit">
                Search
                <span
                  className="a-btn_icon a-btn_icon__on-right cf-icon cf-icon__after cf-icon-search" />
              </button>
              <div
                className="advanced-container">
                <a
                  className="a-btn a-btn__link o-expandable_cue-open"
                  onClick={[Function]}>
                  Show advanced search tips
                </a>
              </div>
            </div>
          </form>
        </nav>
      </div>
    </div>
    <aside
      className="content_sidebar">
      <section
        className="filter-panel">
        <h3>
          Filter results by...
        </h3>
        <section
          className="single-checkbox">
          <h5>
            Only show complaints with narratives?
          </h5>
          <div
            className="m-form-field m-form-field__checkbox">
            <input
              checked={false}
              className="a-checkbox"
              id="theCheckbox"
              onClick={[Function]}
              type="checkbox"
              value="has_narrative" />
            <label
              className="a-label"
              htmlFor="theCheckbox">
              Yes
            </label>
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation date-filter">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Date CFPB Received the complaint
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p />
          <div
            className="layout-row">
            <div
              className="flex-all">
              <label
                className="a-label a-label__heading body-copy">
                From
              </label>
              <div>
                <input
                  aria-describedby="input-error_message-from"
                  className="a-text-input"
                  max={null}
                  min={2014-01-01T17:00:00.000Z}
                  onChange={[Function]}
                  placeholder="mm/dd/yyyy"
                  size="15"
                  value="" />
              </div>
            </div>
            <div
              className="flex-all">
              <label
                className="a-label a-label__heading body-copy">
                Through
              </label>
              <div>
                <input
                  aria-describedby="input-error_message-through"
                  className="a-text-input"
                  max={null}
                  min={2014-01-01T17:00:00.000Z}
                  onChange={[Function]}
                  placeholder="mm/dd/yyyy"
                  size="15"
                  value="" />
              </div>
            </div>
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Matched Company Name
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The company name as it appears in our complaint system, which may be different than the name the consumer provided in their complaint
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <input
              autoComplete="off"
              className="a-text-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter company name"
              type="text"
              value="" />
          </section>
          <ul />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Product / sub-product
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The type of product and sub-product the consumer identified in the complaint
          </p>
          <p>
            <a
              className="a-btn__link"
              href="http://files.consumerfinance.gov/f/documents/201704_cfpb_Summary_of_Product_and_Sub-product_Changes.pdf"
              target="_blank">
              Recent changes to products and sub-products
            </a>
          </p>
          <div>
            <ul />
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Issue / sub-issue
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The type of issue and sub-issue the consumer identified in the complaint
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <input
              autoComplete="off"
              className="a-text-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter name of issue"
              type="text"
              value="" />
          </section>
          <div>
            <ul />
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              State
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The state of the mailing address provided by the consumer
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <input
              autoComplete="off"
              className="a-text-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter state name or abbreviation"
              type="text"
              value="" />
          </section>
          <ul />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Zip Code
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The mailing ZIP code provided by the consumer
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <input
              autoComplete="off"
              className="a-text-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter first three digits of zip code"
              type="text"
              value="" />
          </section>
          <ul />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Did company provide a timely response?
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Company Response
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p>
            How the company responded to the complaint
          </p>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Company Public Response
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p>
            The company\'s optional public-facing response to a consumer\'s complaint
          </p>
        </section>
        <hr />
        <section
          className="o-expandable aggregation date-filter">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Date complaint sent to company
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p />
          <div
            className="layout-row">
            <div
              className="flex-all">
              <label
                className="a-label a-label__heading body-copy">
                From
              </label>
              <div>
                <input
                  aria-describedby="input-error_message-from"
                  className="a-text-input"
                  max={null}
                  min={2014-01-01T17:00:00.000Z}
                  onChange={[Function]}
                  placeholder="mm/dd/yyyy"
                  size="15"
                  value="" />
              </div>
            </div>
            <div
              className="flex-all">
              <label
                className="a-label a-label__heading body-copy">
                Through
              </label>
              <div>
                <input
                  aria-describedby="input-error_message-through"
                  className="a-text-input"
                  max={null}
                  min={2014-01-01T17:00:00.000Z}
                  onChange={[Function]}
                  placeholder="mm/dd/yyyy"
                  size="15"
                  value="" />
              </div>
            </div>
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              Consumer Consent
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p>
            Whether a consumer opted to publish their compaint narrative
          </p>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h5
              className="o-expandable_header-left o-expandable_label">
              How did the consumer submit the complaint to the CFPB?
            </h5>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p />
        </section>
      </section>
    </aside>
    <section
      className="results-panel content_main">
      <summary
        className="action-bar">
        <div>
          <h2>
            ShowingÂ 
            <span>
              0
            </span>
            Â total complaints
          </h2>
        </div>
        <div
          className="layout-row">
          <div
            className="cf-select">
            <select
              id="choose-size"
              onChange={[Function]}
              value={25}>
              <option
                value={10}>
                Show 
                10
                 results
              </option>
              <option
                value={25}>
                Show 
                25
                 results
              </option>
              <option
                value={50}>
                Show 
                50
                 results
              </option>
              <option
                value={100}>
                Show 
                100
                 results
              </option>
            </select>
          </div>
          <div
            className="cf-select">
            <select
              id="choose-sort"
              onChange={[Function]}
              value="created_date_desc">
              <option
                value="created_date_desc">
                Sort by newest to oldest
              </option>
              <option
                value="created_date_asc">
                Sort by oldest to newest
              </option>
              <option
                value="relevance_desc">
                Sort by relevance
              </option>
              <option
                value="relevance_asc">
                Sort by relevance (asc)
              </option>
            </select>
          </div>
          <h5
            className="flex-all">
            <button
              className="a-btn a-btn__link"
              onClick={[Function]}>
              Export results
            </button>
          </h5>
        </div>
      </summary>
      <h2>
        No results were found for your search
      </h2>
      <section
        className="light-box">
        <div
          className="loading-box">
          <span
            className="cf-icon cf-icon-update cf-icon__before cf-icon__spin" />
          <span>
            This page is loading
          </span>
        </div>
      </section>
    </section>
  </div>
  <span />
</main>
`;
